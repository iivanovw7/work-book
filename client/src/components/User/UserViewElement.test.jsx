import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import ViewElement from './UserViewElement';

jest.mock('../../config/apiURL', () => () => '/api');

describe('Testing Button component', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    title: 'title',
    value: 'value'
  };

  const Composition = () => (
    <ViewElement userAccess {...props} />
  );

  it('Should match snapshot', () => {
    const component = mount(<Composition />);
    expect(component).toMatchSnapshot();
  });

  it('Render component with correct props', () => {
    const component = mount(<Composition />);
    expect(component.find(ViewElement).prop('value')).toBe('value');
  });

  it('Should render components with children', () => {
    const component = mount(<Composition />);

    expect(component.find('p')).toHaveLength(1);
    expect(component.find('strong')).toHaveLength(1);

    expect(component.find('p').text()).toBe('title: value');
    expect(component.find('strong').text()).toBe('title: ');
  });
});
