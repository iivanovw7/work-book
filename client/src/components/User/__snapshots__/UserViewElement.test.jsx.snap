// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Should match snapshot 1`] = `ReactWrapper {}`;
